### PR TITLE
Feature/fix post meta

### PR DESCRIPTION
**Pull request by carlosesilva**
The plugin was using the incorrect parameter signature for `update_post_meta` in `do_update_postmeta`. After fixing it I noticed that the `do_update_postmeta` became nothing more than a useless wrapper around `update_post_meta` so I removed it and updated the code to use `update_post_meta` directly.

I also updated the post meta logic inside `has_insecure_content` since it was checking if `$urls` was false but `get_post_meta` actually returns an empty string when the post meta doesn't exist. To be even more safe I changed it to use the `metadata_exists` function.

### CODE REVIEW

**Comment by carlosesilva** [comment]
[BU-SSL](https://trello.com/c/WFhMr2Q2/10-bu-ssl)

File: bu-ssl.php [hunk 1]
```diff
@@ -272,20 +270,37 @@ public function search_for_insecure_content( $content, $type = 'any' ) {
...
+			// Get the post object from the post id.
+			$post = get_post( $post_id );
+
+			// Search for insecure content.
+			$urls = self::search_for_insecure_content( $post->post_content, $search_type );
 
-		if ( false === $urls ) {
-			$urls = self::search_for_insecure_content( $content, $search_type );
-			self::do_update_postmeta( $meta_key, $post_id, $urls );
+			// Update post meta.
```
Comment (Adam42) [inline bu-ssl.php]: The `// Update post meta. `comment is superfluous since the function is called `update_post_meta`

File: bu-ssl.php [hunk 2]
```diff
@@ -338,31 +353,19 @@ public function update_post( $post_id ) {
 		$urls = self::search_for_insecure_content( $post->post_content );
 
 		// Update post meta with list of insecure urls.
-		self::do_update_postmeta( $this->options['post_meta_key'], $post_id, $urls );
+		update_post_meta( $post_id, $this->options['post_meta_key'], $urls );
```
Comment (Adam42) [inline bu-ssl.php:355]: Great change, especially as `do_update_postmeta` swaps the param order for `update_post_meta` which is real confusing!

**Review by Adam42 — APPROVED** [review]
Looks good to me!

**Pull request merged**